src/java/fr/paris/lutece/portal/service/securityheader/SecurityHeaderService.java

```diff
@@ -0,0 +1,300 @@
+	 * 
+	 * @return collection of security headers
+	 */
+	public Collection<SecurityHeader> findAll( )
+    {
+		Collection<SecurityHeader> securityHeadersList = null;
+		
+		if( _bMapRefreshDone )
+		{
+			securityHeadersList = _mapSecurityHeaders.values( );
+		} 
+		else
+		{			
+			securityHeadersList = SecurityHeaderHome.findAll( );
+			refreshSecurityHeadersMap( securityHeadersList );	
+		}
+		
+		return securityHeadersList;
+    }
+	
+	/**
+	 * Returns a list of all security headers used in manage security headers page. They are sorted by type, page category and name. 
+	 * Type and Page category wordings are filled in order to provide a more user-friendly text in the page.
+	 * 
+	 * @param locale
+	 *           The locale
+	 * @return list of all security headers
+	 */
+	public List<SecurityHeader> findAllForDisplay( Locale locale )
+    {
+		List<SecurityHeader> securityHeadersList = findAll( ).stream().collect( Collectors.toList( ) );
+    	Map<String, String> categoryPageMap = getPageCategoryList( locale ).toMap( );
+    	Map<String, String> typeMap = getTypeList( locale ).toMap( );
+    	
+    	for( SecurityHeader securityHeader : securityHeadersList )
+    	{
+    		securityHeader.setTypeWording( typeMap.get( securityHeader.getType( ) ) );
+    		securityHeader.setPageCategoryWording( categoryPageMap.get( securityHeader.getPageCategory( ) ) );   		
+    	}
+    	
+		Collections.sort( securityHeadersList, Comparator.comparing( SecurityHeader::getType )
+				                                         .thenComparing( SecurityHeader::getPageCategory, Comparator.nullsLast( Comparator.naturalOrder( ) ) )
+				                                         .thenComparing( SecurityHeader::getName ) );
+    	
+		return securityHeadersList;
+    }
+	
+	/**
+	 * Returns the reference list of security headers types
+	 * 
+	 * @param locale
+	 *           The locale
+	 * @return ReferenceList object with security headers types
+	 */
+	public ReferenceList getTypeList( Locale locale )
+	{
+		ReferenceList listTypes = new ReferenceList( );
+		
+		listTypes.addItem( SecurityHeaderType.PAGE.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_TYPE, locale ) );
+		listTypes.addItem( SecurityHeaderType.REST_API.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_REST_API_TYPE, locale ) );
+		
+		return listTypes;
+	}
+	
+	/**
+	 * Returns the reference list of security headers page categories
+	 * 
+	 * @param locale
+	 *           The locale
+	 * @return ReferenceList object with security headers page categories
+	 */
+	public ReferenceList getPageCategoryList( Locale locale )
+	{
+		ReferenceList listPageCategory = new ReferenceList( );
+		
+		listPageCategory.addItem( SecurityHeaderPageCategory.ALL.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ALL, locale ) );		
```

Comment:
Taken into account in commit 9998e65.

In the first place, the solution above has been implemented but after reading your mail sent on Thursday 20, I have read the Lutece wiki and it gave me an idea to simplify how to get the translations easily. This is this solution that has been committed :

- Type/categories labels have been renamed in securityheader_messages.properties and securityheader_messages_fr.properties files so that the last element corresponds to the code of the type/category that we found in enums SecurityHeaderType/SecurityHeaderCategory.
For instance, the label of rest API type was labelRestApiType REST that became type.label.rest_api and the last element is rest_api.

-  In security header creation and modification templates, labels are retrieved like that : 
#i18n{${"portal.securityheader.type.label."+type.code}} for types
#i18n{${"portal.securityheader.category.label."+category.code}} for categories

- In security headers list template, labels are retrieved like that :
#i18n{${"portal.securityheader.type.label."+"${security_header.type}"}} for types
#i18n{${"portal.securityheader.category.label."+"${security_header.pageCategory}"}} for categories